stake-pool/single-pool-cli/tests/test.rs

```diff
@@ -0,0 +1,284 @@
+use {
+    solana_cli_config::Config as SolanaConfig,
+    solana_sdk::{
+        bpf_loader_upgradeable,
+        native_token::LAMPORTS_PER_SOL,
+        pubkey::Pubkey,
+        signature::{write_keypair_file, Keypair, Signer},
+        stake::{
+            self,
+            state::{Authorized, Lockup, StakeState},
+        },
+        system_instruction, system_program,
+        transaction::Transaction,
+    },
+    solana_test_validator::{TestValidator, TestValidatorGenesis, UpgradeableProgramInfo},
+    solana_vote_program::{
+        vote_instruction::{self, CreateVoteAccountConfig},
+        vote_state::{VoteInit, VoteState, VoteStateVersions},
+    },
+    spl_single_validator_pool::{self as single_pool},
+    spl_token_client::client::{ProgramClient, ProgramRpcClient, ProgramRpcClientSendTransaction},
+    std::{path::PathBuf, process::Command, str::FromStr, sync::Arc},
+    tempfile::NamedTempFile,
+};
+
+// XXX TODO FIXME uhh few things... need to ensure binary is built
+// also i dont know how to advance an epoch on TestValidator?
+// i guess its just running the fill binary? so maybe thats actually impossible?
```

Comment:
The way we do this now is by setting a very short epoch, ie 32 slots, then wait until the epoch rolls over. Here's the example from the stake bot tests:

* short epochs: https://github.com/solana-foundation/stake-o-matic/blob/9e90e1c86f4ed82a5e6cba617000f49ca54fe762/bot/src/stake_pool_v0.rs#L937-L943
* waiting https://github.com/solana-foundation/stake-o-matic/blob/9e90e1c86f4ed82a5e6cba617000f49ca54fe762/bot/src/rpc_client_utils.rs#L517